tests/30rooms/13anonymousaccess.pl

```diff
@@ -0,0 +1,96 @@
+test "Anonymous user cannot view non-world-readable rooms",
+   requires => [ qw( first_api_client ), local_user_preparer() ],
+
+   do => sub {
+      my ( $api_client, $user ) = @_;
+
+      my $anonymous_user;
+      my $room_id;
+
+      register_anonymous_user( $api_client )->then( sub {
+         ( $anonymous_user ) = @_;
+
+         matrix_create_and_join_room( [ $user ] )
+         ->then( sub {
+            ( $room_id ) = @_;
+
+            do_request_json_for( $user,
+               method => "PUT",
+               uri    => "/api/v1/rooms/$room_id/state/m.room.history_visibility/",
+
+               content => {
+                  history_visibility => "shared",
+               },
+            );
+         })->then( sub {
+            matrix_send_room_text_message( $user, $room_id, body => "mice" )
+         })->then( sub {
+            do_request_json_for( $anonymous_user,
+               method => "GET",
+               uri => "/api/v1/rooms/${room_id}/messages",
+               params => {
+                  limit => "1",
+                  dir => "b",
+               },
+            )
+         })->followed_by(\&main::expect_http_403);
```

Comment:
We don't seem to need a "followed_by" in any other of the places we use main::expect_http_403 in the tests.
Is this necessary?